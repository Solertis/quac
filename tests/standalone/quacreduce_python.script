#!/bin/bash

# Test that the Python interface of QUACreduce works.
#
# Copyright (c) 2012-2013 Los Alamos National Security, LLC, and others.

. ./environment.sh

cd $DATADIR


## Set up input

y "echo -e 'foo bar baz\nfoo foo' > foo1.txt"
y "echo -e 'bar' > foo2.txt"
y "cat foo*.txt"


## Do map-reduce

y "quacreduce --python qr.wordcount.Job --pyargs 'factor:2' foo*.txt"
x make --quiet  # output contains temp dirs that vary
y "cat out/* | sort"
