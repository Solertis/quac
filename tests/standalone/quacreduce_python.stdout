$ (echo -e 'foo bar baz\nfoo foo' > foo1.txt)
$ (echo -e 'bar' > foo2.txt)
$ (cat foo*.txt)
foo bar baz
foo foo
bar
$ (quacreduce --python qr.wordcount.Job --pyargs 'factor:2' foo*.txt)
$ make --quiet
$ (cat out/* | sort)
2 baz
4 bar
6 foo
